File: java/1.7/factory/src/com/pizza/pizzaIngredientFactory/NYPizzaIngredientFactory.java
package com.pizza.pizzaIngredientFactory;

import com.pizza.cheese.Cheese;
import com.pizza.cheese.ReggianoCheese;
import com.pizza.clams.Clams;
import com.pizza.clams.FreshClams;
import com.pizza.dough.Dough;
import com.pizza.dough.ThinCrustDough;
import com.pizza.pepperoni.Pepperoni;
import com.pizza.pepperoni.SlicedPepperoni;
import com.pizza.sauce.MarinaraSauce;
import com.pizza.sauce.Sauce;
import com.pizza.veggies.Garlic;
import com.pizza.veggies.Mushroom;
import com.pizza.veggies.Onion;
import com.pizza.veggies.RedPepper;
import com.pizza.veggies.Veggies;

public class NYPizzaIngredientFactory implements PizzaIngredientFactory {
 
	public Dough createDough() {
		return new ThinCrustDough();
	}
 
	public Sauce createSauce() {
		return new MarinaraSauce();
	}
 
	public Cheese createCheese() {
		return new ReggianoCheese();
	}
 
	public Veggies[] createVeggies() {
		Veggies veggies[] = { new Garlic(), new Onion(), new Mushroom(), new RedPepper() };
		return veggies;
	}
 
	public Pepperoni createPepperoni() {
		return new SlicedPepperoni();
	}

	public Clams createClam() {
		return new FreshClams();
	}
}

File: python/2.7/factory/pizzaIngredientFactory/NYPizzaIngredientFactory.py
import sys

from PizzaIngredientFactory import *

sys.path.append("./dough")
from ThinCrustDough import *

sys.path.append("./sauce")
from MarinaraSauce import *

sys.path.append("./cheese")
from ReggianoCheese import *

sys.path.append("./veggies")
from Garlic import *
from Onion import *
from Mushroom import *
from RedPepper import *

sys.path.append("./pepperoni")
from SlicedPepperoni import *

sys.path.append("./clams")
from FreshClams import *

class NYPizzaIngredientFactory(PizzaIngredientFactory):
    def __init__(self):
        super(NYPizzaIngredientFactory, self).__init__()

    def createDough(self):
        return str(ThinCrustDough())

    def createSauce(self):
        return str(MarinaraSauce())

    def createCheese(self):
        return str(ReggianoCheese())

    def createVeggies(self):
        self.veggies = [str(Garlic()), str(Onion()), str(Mushroom()), str(RedPepper())]
        return self.veggies

    def createPepperoni(self):
        return str(SlicedPepperoni())

    def createClam(self):
        return str(FreshClams())